# HG changeset patch
# User Jason Evans <jasone@canonware.com>
# Date 1232168582 28800
# Node ID 18643350b1c548c9ec5cdd0721446ddd5c877b9d
# Parent  682a2248596fd367913d0c67d2fbd762d5226606
Do not mangle enum constant names if the enum is public.

diff --git a/Cython/Compiler/Symtab.py b/Cython/Compiler/Symtab.py
--- a/Cython/Compiler/Symtab.py
+++ b/Cython/Compiler/Symtab.py
@@ -303,7 +303,7 @@
     def declare_const(self, name, type, value, pos, cname = None, visibility = 'private'):
         # Add an entry for a named constant.
         if not cname:
-            if self.in_cinclude:
+            if self.in_cinclude or visibility == 'public':
                 cname = name
             else:
                 cname = self.mangle(Naming.enum_prefix, name)
